### PR TITLE
Introduce display tab in raster layer properties

### LINKED ISSUE #7887: Raster maptips (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#50854
Author: @elpaso
QGIS version: 3.30

**Raster maptips**

### PR Description:
MapTips were only available for vector layers, by moving the maptips API from `QgsVectorLayer` to `QgsMapLayer` it becomes possible to implement maptips for other layer types.

This PR adds the implementation for raster layers and a few expression functions and utilities to make it useful:

- `map_to_html_table`: converts a map (key value pair data structure) to an HTML table
- `map_to_html_dl`: converts a map (key value pair data structure) to an HTML definition list
- `raster_attributes`: creates a map with the fields names as keys and the raster attribute table values as values from the attribute table entry that matches the given raster value
- a new "Map Layer Position" expression scope that defines a variable `@layer_cursor_point` with the layer coordinates (in layer's CRS) under the mouse position

An expression that shows raster attributes in a maptip could look like this:

raster_maptips_config

Which renders:

raster_maptips

Funded by: NOAA OCS Hydrography

### Commits tagged with [need-docs] or [FEATURE]

### LINKED ISSUE #8199: Add a way to disable map tips for a specific layer (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#52351
Author: @YoannQDQ
QGIS version: 3.32

**Add a way to disable map tips for a specific layer**

### PR Description:
- fix #23400

## Description

This PR add a `mapTipsEnabled` Q_PROPERTY in the `QgsMapLayer` which allows disabling mapTips for a specific layer.

disable_map_tips

### Commits tagged with [need-docs] or [FEATURE]

### LINKED ISSUE #8210: Map tip preview (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#52420
Author: @YoannQDQ
QGIS version: 3.32

**Map tip preview**

## Description

Adds a map tip preview in the Raster and Vector layer properties dialog.
This makes it easier to design map tips.

map_tip_preview

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #8199, fixes #8210 and fixes #7887
- [x] TODO: add a screenshot with a working code

### CODE REVIEW

**Comment by DelazJ** [comment]
@elpaso Could you share the RAT data you used in the report, please? I think showing such an advanced way to manipulate the fonctions and pixels would be nice. And I'm failing to get anything with raster_attributes on a "simple" raster file.

File: docs/user_manual/expressions/functions_list.rst [hunk 1]
```diff
@@ -1184,6 +1184,7 @@ To use these variables in an expression, they should be preceded by the
    "layer", "The current layer"
    "layer_crs", "The Coordinate Reference System Authority ID of the current layer"
    "layer_crs_ellipsoid", "The ellipsoid Authority ID of the current layer CRS"
+   "layer_cursor_point", "Point geometry under the mouse position in map canvas, in active layer's CRS"
```
Comment (pathmapper) [inline docs/user_manual/expressions/functions_list.rst:1187]:
```suggestion
   "layer_cursor_point", "Point geometry under the mouse position in map canvas (or the GetFeatureInfo position in context of QGIS Server), in active layer's CRS"
```

**Review by pathmapper — COMMENTED** [review]
This should fix https://github.com/qgis/QGIS-Documentation/issues/8769.

File: docs/user_manual/expressions/functions_list.rst (hunk 1 above)
Comment (DelazJ) [reply docs/user_manual/expressions/functions_list.rst:1187]: Well, let's keep that adjustment aside for now and reuse it after merging, as this PR should be backported to 3.34.

**Pull request merged**